src/lib.rs

```diff
@@ -136,6 +136,14 @@ pub struct BlockBasedOptions {
     inner: *mut ffi::rocksdb_block_based_table_options_t,
 }
 
+/// Defines the underlying memtable implementation.
+/// See https://github.com/facebook/rocksdb/wiki/MemTable for more information.
+pub enum MemtableFactory {
+    Vector,
+    HashSkipList { bucket_count: usize, height: i32, branching_factor: i32 },
```

Comment:
wasn't sure if these should be defined with `libc` types or not - let me know what you prefer.

Reply:
I prefer the way you chose, by using rust types where possible for the public API.